src/java/voldemort/server/VoldemortServer.java

```diff
@@ -524,26 +524,28 @@ public void goOffline() {
     }
 
     public void goOnline() {
-        switch (validateReadOnlyStoreStatusBeforeGoingOnline()) {
-            case SOME_STORES_DISABLED:
-                throw new VoldemortException("Aborting the goOnline() request because of disabled Read-Only store(s)! " +
-                        "Voldemort is still offline.");
-            case UNKNOWN_HA_STATE:
-                createOnlineServices();
-                startOnlineServices();
-                throw new VoldemortException("Unable to determine (and potentially clean up) the remote HA state. " +
-                        "Voldemort still went online, but further BnP job failures may not be covered by HA.");
-            case NO_STORES_DISABLED:
-                createOnlineServices();
-                startOnlineServices();
-                break;
+        ReadOnlyStoreStatusValidation validation = validateReadOnlyStoreStatusBeforeGoingOnline();
+
+        if (validation.readyToGoOnline) {
+            createOnlineServices();
+            startOnlineServices();
+            getMetadataStore().setOfflineState(false);
```

Comment:
In the existing code, this was the first statement. Any reason, why you are changing the order ?

If not, I would like to keep the getMetadataStore().setOfflineState(false) to be the first statement.